public namespace uf;
public import core::common -;
public import core::io::errno -;

public union foo {
  public errno_t err;
  public string value;
}

{
  set_stack_trace_limit(0);
  try {
    foo x;
    x.err = EINTR;
    println(x.value); // throws
  } catch (c_exception ex) {
    println("caught", ex);
  }
}

